### ISSUE TITLE
Some MOS data not filtered out 

### ISSUE THREAD

**Issue by Eb-Zeero** [saltastroops/salt-api#41]
Example:
   - Semester 2021-1 have 34 results and enp-poit "/mos" have 37
   - filters Mask 1b, Mask 2b and Mask 3b are only coming up only once on the live but "/mos" have them twice

**Comment by Eb-Zeero** [comment]
Issue fixed
OLD-WM Looks for masks that are not needed for the selected semester and mark then as no longer needed.
NEW-WM looks for mask that are no longer needed for the current semester regardless of the selected semester.
